types/wlr_screencopy_v1.c

```diff
@@ -48,18 +48,18 @@ static void screencopy_damage_accumulate(struct screencopy_damage *damage) {
 		return;
 	}
 
+	int width, height;
+	wlr_output_transformed_resolution(output, &width, &height);
```

Comment:
This looks like it will break stuff.

Reply:
Ah indeed, damage is accumulated in transformed coordinates, but when it is committed, it is in buffer coordinates ...

Reply:
Ah, interesting. I suppose we should just revert to `output->{width,height}` then?

I'll also create an issue about removing all of this mess.

Reply:
I haven't confirmed that it breaks anything. I just thought this looked suspicious because I've never run into an issue with damage reporting on transformed outputs before. I dunno, maybe it fixes bugs that I didn't know about.

Reply:
I don't see any damage conversion in `wlr_output_set_damage` nor in `wlr_output_commit`. So I believe `wlr_output_state.damage` is in buffer-local coords. The comment in the header file even says so:

```c
struct wlr_output_state {
	pixman_region32_t damage; // output-buffer-local coordinates
```

In fact, Sway already transforms the damage manually before calling `wlr_output_set_damage`:

```c
	enum wl_output_transform transform =
		wlr_output_transform_invert(wlr_output->transform);
	wlr_region_transform(&frame_damage, &output->damage->current,
		transform, width, height);
	wlr_output_set_damage(wlr_output, &frame_damage);
	pixman_region32_fini(&frame_damage);
```

tl;dr we should use `output->{width,height}` here.

Reply:
https://github.com/swaywm/wlroots/issues/2518

Reply:
#2519